astropy/utils/tests/test_data.py

```diff
@@ -2138,7 +2138,6 @@ def test_clear_download_cache_variants(temp_cache, valid_urls):
     assert not is_url_in_cache(u)
 
 
-@pytest.mark.skipif("CI", reason="Flaky on CI")
```

Comment:
pytest-rerunfailure maybe an answer for real flaky stuff, assuming the timescale of the flakyness is not hours or days but for consecutive runs. 
See astroquery for example usage.

Reply:
I actually am really not so sure how meaningful it is to spam the IERS A download site with repeated requests for what are not small files. This might make more sense on monthly cron.

Reply:
p.s. But for *this* PR I would suggest to just leave the `skipif` in place and get your nice solution in!

Reply:
Yes, the download takes not-insignificant time, so putting it into weekly or monthly cron would make more sense – but the only one using remote data is py310-test-alldeps-predeps in cron_daily, it seems. Perhaps there also is a smaller file we could use?

Reply:
> – but the only one using remote data is `py310-test-alldeps-predeps` in cron_daily, it seems.

Except, is it really? It has `toxposargs: --remote-data=any` (ci_workflows have `posargs: --remote-data=foo`), but is not using any remote-data flags and the [last daily](https://github.com/astropy/astropy/actions/runs/3598502712/jobs/6061324996#step:9:198) was running with 
```
Using Astropy options: remote_data: none.

ARCH_ON_CI: undefined
IS_CRON: undefined
```
@pllim do you know if `os.environ.get("CI")` should be set there?

Reply:
I think "CI" env var is set up by Actions. It should always be true in CI setting.

FWIW, @saimn also put in the `hugemem` flag but we currently do not enable it on CI.

https://github.com/astropy/astropy/blob/252055ef3b9d6be3dd0e7ab3e7b70e1f28b80421/astropy/io/ascii/tests/test_c_reader.py#L1157

Reply:
Ah, yes, here is the documentation from Actions: https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables

Reply:
I propose we add the skipif back in and open up a follow-up issue. I don't think we need to really test this in backport branch, but only going forward in main?

I agree a cron to actually test this is good.

Reply:
Might wanna squash too? 😸

Reply:
> Might wanna squash too? 😸

Just replaced the other commit 😏 – it is skipped in the workflows now, although conftest does not report the `IS_CRON` value correctly, and it won't run in cron_daily while remote_data are not enabled.
I wonder if the variables set in `env` are passed through the OpenAstronomy workflow; and in particular found it lists `toxargs` and `posargs` as inputs, but not `toxposargs`. But that will be another issue indeed!

Reply:
Oh, right... OpenAstronomy does not pass custom env var... See https://github.com/astropy/astropy/pull/14070#discussion_r1035351968 . But I reverted the weekly cron to not use it in https://github.com/astropy/astropy/pull/14077 . So theoretically, this check means this test would run on weekly cron only, so we might need to change this?

https://github.com/astropy/astropy/blob/252055ef3b9d6be3dd0e7ab3e7b70e1f28b80421/.github/workflows/ci_cron_weekly.yml#L42-L43

Reply:
Yes, and weekly would probably be enough – I did not quite understand why remote was disabled there, but good for daily anyway.

Reply:
@bsipocz , rerunfailure does look promising for some remote data tests we have here, but that is a battle for another day. Thanks for the suggestion!